tests/src/test/java/alluxio/master/file/ConcurrentDeleteIntegrationTest.java

```diff
@@ -78,19 +74,10 @@
           "sleep://" + mLocalUfsPath).setProperty(PropertyKey
           .USER_FILE_MASTER_CLIENT_THREADS, CONCURRENCY_FACTOR).build();
 
-  // Must be done in beforeClass so execution is before rules
-  @BeforeClass
-  public static void beforeClass() throws Exception {
-    SleepingUnderFileSystemOptions options = new SleepingUnderFileSystemOptions();
-    sSleepingUfsFactory = new SleepingUnderFileSystemFactory(options);
-    options.setDeleteFileMs(SLEEP_MS).setDeleteDirectoryMs(SLEEP_MS);
-    UnderFileSystemFactoryRegistry.register(sSleepingUfsFactory);
-  }
-
-  @AfterClass
-  public static void afterClass() throws Exception {
-    UnderFileSystemFactoryRegistry.unregister(sSleepingUfsFactory);
-  }
+  @Rule
```

Comment:
For rule in this file, instead of using `@Rule`, please use `@ClassRule`. The difference is that `@Rule` is set and restored for each single test case (matching `@Before` and `@After`), while `@ClassRule` is set and restored for each test class (possibly across multiple test cases). Since the register and unregister happens in `@BeforeClass` and `@AfterClass`, we should use `@ClassRule`